lucene/codecs/src/java/org/apache/lucene/codecs/memory/DirectPostingsFormat.java

```diff
@@ -0,0 +1,2270 @@
+      final int termLen = termOffsets[1+ord] - upto;
+      int otherUpto = other.offset;
+
+      final int stop = upto + Math.min(termLen, other.length);
+      while (upto < stop) {
+        int diff = (termBytes[upto++] & 0xFF) - (otherBytes[otherUpto++] & 0xFF);
+        if (diff != 0) {
+          return diff;
+        }
+      }
+
+      // One is a prefix of the other, or, they are equal:
+      return termLen - other.length;
+    }
+
+    private void setSkips(int termOrd, byte[] termBytes) {
+
+      final int termLength = termOffsets[termOrd+1] - termOffsets[termOrd];
+
+      if (sameCounts.length < termLength) {
+        sameCounts = ArrayUtil.grow(sameCounts, termLength);
+      }
+
+      // Update skip pointers:
+      if (termOrd > 0) {
+        final int lastTermLength = termOffsets[termOrd] - termOffsets[termOrd-1];
+        final int limit = Math.min(termLength, lastTermLength);
+
+        int lastTermOffset = termOffsets[termOrd-1];
+        int termOffset = termOffsets[termOrd];
+
+        int i = 0;
+        for(;i<limit;i++) {
+          if (termBytes[lastTermOffset++] == termBytes[termOffset++]) {
+            sameCounts[i]++;
+          } else {
+            for(;i<limit;i++) {
+              if (sameCounts[i] >= minSkipCount) {
+                // Go back and add a skip pointer:
+                saveSkip(termOrd, sameCounts[i]);
+              }
+              sameCounts[i] = 1;
+            }
+            break;
+          }
+        }
+
+        for(;i<lastTermLength;i++) {
+          if (sameCounts[i] >= minSkipCount) {
+            // Go back and add a skip pointer:
+            saveSkip(termOrd, sameCounts[i]);
+          }
+          sameCounts[i] = 0;
+        }
+        for(int j=limit;j<termLength;j++) {
+          sameCounts[j] = 1;
+        }
+      } else {
+        for(int i=0;i<termLength;i++) {
+          sameCounts[i]++;
+        }
+      }
+    }
+
+    private void finishSkips() {
+      assert count == terms.length;
+      int lastTermOffset = termOffsets[count-1];
+      int lastTermLength = termOffsets[count] - lastTermOffset;
+
+      for(int i=0;i<lastTermLength;i++) {
+        if (sameCounts[i] >= minSkipCount) {
+          // Go back and add a skip pointer:
+          saveSkip(count, sameCounts[i]);
+        }
+      }
+
+      // Reverse the skip pointers so they are "nested":
+      for(int termID=0;termID<terms.length;termID++) {
+        TermAndSkip term = terms[termID];
+        if (term.skips != null && term.skips.length > 1) {
+          for(int pos=0;pos<term.skips.length/2;pos++) {
+            final int otherPos = term.skips.length-pos-1;
+
+            final int temp = term.skips[pos];
+            term.skips[pos] = term.skips[otherPos];
+            term.skips[otherPos] = temp;
+          }
+        }
+      }
+    }
+
+    private void saveSkip(int ord, int backCount) {
+      final TermAndSkip term = terms[ord - backCount];
+      skipCount++;
+      if (term.skips == null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of == on line 636 to != check before setting the field value on line 645 or verify if the check is needed at all.